compiler/src/lib.rs

```diff
@@ -144,6 +162,19 @@ fn compile<T: FieldElement>(
             &commits,
         );
         log::info!("Wrote commits.bin.");
+        match prove_with {
+            #[cfg(feature = "halo2")]
+            Some(Backend::Halo2) => {
+                use std::io::Write;
+                let proof = halo2::prove_ast(analyzed, constants, commits);
+                let mut proof_file = fs::File::create(output_dir.join("proof.bin")).unwrap();
```

Comment:
Maybe the proof's filename should also come from the user? Also fine doing that later.